hack/make/.integration-test-helpers

```diff
@@ -40,6 +42,20 @@ go_test_dir() {
 	)
 }
 
+ensure_test_dir() {
+	# make sure a test dir will compile
+	dir="$1"
+	out="$2"
+	if ! (
+		set -e
+		cd "$dir"
+		echo Building test dir: "$dir"
+		go test -c -o "$out" -ldflags "$LDFLAGS" "${BUILDFLAGS[@]}"
+	); then
```

Comment:
How does this if/then/fi change the behaviour ? Won't this already exit because of `set -e` ?